File: packages/python/cronixui/empty_state.py
"""CronixUI EmptyState Component - Native tkinter implementation."""
import tkinter as tk
from .core import Frame, get_theme


class EmptyState(Frame):
    """Empty state placeholder with icon, title, and description."""
    def __init__(self, master, icon="", title="", description="", **kwargs):
        super().__init__(master, bg=get_theme().bg, highlightthickness=0, **kwargs)
        theme = get_theme()
        if icon:
            tk.Label(self, text=icon, bg=theme.bg, font=(theme.font_family, 28)).pack(pady=(0, 8))
        tk.Label(self, text=title, bg=theme.bg, fg=theme.text,
                font=(theme.font_family, 14, 'bold')).pack()
        if description:
            tk.Label(self, text=description, bg=theme.bg, fg=theme.text_muted,
                    font=(theme.font_family, 10), wraplength=400).pack(pady=(4, 8))

File: packages/rust/cronixui/src/components/empty_state.rs
//! EmptyState component

/// Empty state placeholder.
pub struct EmptyState {
    pub title: String,
    pub description: Option<String>,
    pub icon: Option<String>,
}

impl EmptyState {
    pub fn new(title: &str) -> Self {
        Self { title: title.to_string(), description: None, icon: None }
    }
    pub fn with_description(mut self, desc: &str) -> Self { self.description = Some(desc.to_string()); self }
    pub fn with_icon(mut self, icon: &str) -> Self { self.icon = Some(icon.to_string()); self }
}

impl Default for EmptyState {
    fn default() -> Self { Self::new("") }
}
